tests/AzureBlobStorageAdapterTest.php

```diff
@@ -0,0 +1,480 @@
+    ////                        ],
+    ////                    ],
+    ////                    "NextMarker" => null,
+    ////                ]
+    ////            ]));
+    ////        $blobClient = BlobRestProxy::createBlobService('AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1;TableEndpoint=http://azurite:10002/devstoreaccount1;');
+    //        $service = new AzureBlobStorageAdapter($blobClient, $this->logger, 'default');
+    //        $response = $service->listContents('', true);
+    //        $this->assertIsIterable($response, 'Error getting the list of the content');
+    //    }
+
+    /**
+     * @dataProvider getCopyFiles
+     */
+    public function testCopy($exception)
+    {
+        $blobClient = $this->createMock(BlobRestProxy::class);
+        if ($exception) {
+            $blobClient->expects(self::any())
+                ->method('copyBlob')
+                ->willThrowException(new UnableToCopyFile(""));
+            $this->expectException(UnableToCopyFile::class);
+        } else {
+            $blobClient->expects(self::any())
+                ->method('copyBlob')
+                ->willReturn(null);
+            $this->expectNotToPerformAssertions();
+        }
+
+        //        $blobClient = BlobRestProxy::createBlobService('AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1;TableEndpoint=http://azurite:10002/devstoreaccount1;');
+        $service = new AzureBlobStorageAdapter($blobClient, $this->logger, 'default');
+        $service->copy(self::FILE_TEST, 'test.png', new Config());
+    }
+
+    /**
+     * @dataProvider getCopyFiles
+     */
+    public function testMove($exception)
+    {
+        $blobClient = $this->createMock(BlobRestProxy::class);
+        if ($exception) {
+            $blobClient->expects(self::any())
+                ->method('copyBlob')
+                ->willThrowException(new UnableToCopyFile(""));
+            $this->expectException(UnableToCopyFile::class);
+        } else {
+            $blobClient->expects(self::any())
+                ->method('copyBlob')
+                ->willReturn(null);
+            $this->expectNotToPerformAssertions();
+        }
+
+        //        $blobClient = BlobRestProxy::createBlobService('AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;BlobEndpoint=http://azurite:10000/devstoreaccount1;QueueEndpoint=http://azurite:10001/devstoreaccount1;TableEndpoint=http://azurite:10002/devstoreaccount1;');
```

Comment:
Is this code line needed?

Reply:
only about how to test it locally with the service